gapic/src/main/com/google/gapid/perfetto/views/TraceComposite.java

```diff
@@ -200,6 +208,7 @@ public TraceComposite(Composite parent, Analytics analytics, Theme theme) {
 
   protected abstract S createState();
   protected abstract RootPanel<S> createRootPanel();
+  protected abstract FilterDialog createFilterDialog();
```

Comment:
We don't actually expect FilterDialog to be subclassed, so I don't really understand this abstract method. If you want subclasses to be able to customize the TopBar, create an abstract function that passes it in, or create a method returning a boolean of whether to have a filter dialog or not.